kernel_gateway/services/notebooks/handlers.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+import tornado.web
+import json
+from queue import Empty
+from .request_utils import *
+
+class NotebookAPIHandler(tornado.web.RequestHandler):
+    get_source = None
+    put_source = None
+    post_source = None
+    delete_source = None
+    kernel_client = None
+    execution_timeout = 5
+    def initialize(self, sources, kernel_client):
```

Comment:
At the moment, the init_webapp function of gatewayapp.py is sending the parsed cell source and URI to Tornado when registering the hander class, so however messy it is right now, we're only parsing things once.